astroquery/mpc/core.py

```diff
@@ -1061,12 +1059,20 @@ def _parse_result(self, result, **kwargs):
             # find column headings
             if SKY:
                 # slurp to newline after "h m s"
-                i = text_table.index('\n', text_table.index('h m s')) + 1
+                # raise EmptyResponseError if no ephemeris lines are found in the query response
+                try:
+                    i = text_table.index('\n', text_table.index('h m s')) + 1
+                except ValueError as e:
+                    raise NoResultsWarning(content)
                 columns = text_table[:i]
                 data_start = columns.count('\n') - 1
             else:
                 # slurp to newline after "JD_TT"
-                i = text_table.index('\n', text_table.index('JD_TT')) + 1
+                # raise EmptyResponseError if no ephemeris lines are found in the query response
+                try:
+                    i = text_table.index('\n', text_table.index('JD_TT')) + 1
+                except ValueError as e:
```

Comment:
Please add a test that covers this exception.

Reply:
@mkelley 
It is difficult to create a remote test for this case as I can no longer repeate this issue by querying the MPC for the same object and date that i described in the issu (looks like something was not working on the MPC side at that time). I'm not able to find a different target or date where such a response would be returned. But if it happens I can add a remote test for this case also.

So I have manually created the test data file by removing the ephemeris lines from the result table and included a test for this in local data test_mpc.py
```
def test_get_ephemeris_by_name_empty(patch_post):
    with pytest.raises(NoResultsWarning):
        mpc.core.MPC.get_ephemeris('340P', location='G37')
```

Reply:
I think the manual test is OK.  I can't think of a successful remote test case that **should** return empty lines.    Brigitta's impactor example is an interesting idea, but the service clearly doesn't behave that way.

Reply:
>  I can't think of a successful remote test case that should return empty line

well, that case I would revert my previous opinion about the warning and say that it should loudly error as an indication of something is broken.

Reply:
@mkelley, @bsipocz 
So we change back to EmptyResponseError?

Reply:
I agree, EmptyResponseError

Reply:
@mkelley 
I have corrected it back to EmptyResponseerror